### PR TITLE
[WIP] TestingProposals-241

### PR DESCRIPTION
**Pull request by ayuryshev**
I'm publishing right now results to review and discuss

1. TestingProposals.md - the doc which I started, personally I think
that second doc is better for us - it's more specific for skywire
2. PeriodicTableOfTests.md - it's more concentrated on measurement and
decision support
3. periodic-table-of-tests.zsh - is a POC, how to create measurements. It's only for illustration purposes

Closes: #241

### CODE REVIEW

File: TestingProposals.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+## 1.  Tests in  development environment and automated tests
+
+### Description
+
+**Developer environment**:
+
+- very simplistic but allows fast refresh-restart
+- it could be 2-3 nodes  in one-two networks
+- easily integrated with IDEs, etc
```
Comment (evanlinjin) [inline TestingProposals.md:68]: Not priority.

File: TestingProposals.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+-------------------------------------------------------------
+
+## 1.  Tests in  development environment and automated tests
+
+### Description
+
+**Developer environment**:
+
+- very simplistic but allows fast refresh-restart
+- it could be 2-3 nodes  in one-two networks
```
Comment (evanlinjin) [inline TestingProposals.md:67]: We need to be able to easily configure the number of nodes/services for the local network.

File: TestingProposals.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+- absolute logical consistency
+- full coverage of the theme of software testing
+
+-------------------------------------------------------------
+
+## 1.  Tests in  development environment and automated tests
+
+### Description
+
+**Developer environment**:
```
Comment (evanlinjin) [inline TestingProposals.md:64]: There should be two types of developer environments:
1. Local environment (all services and the entire Skywire network runs locally).
2. Public environment (all services except `skywire-node` runs on public servers).

File: periodic-table-of-tests.zsh [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/bin/zsh
```
Comment (evanlinjin) [inline periodic-table-of-tests.zsh:1]: Not needed.

File: PeriodicTableOfTests.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
+# "Periodic Table of Tests"
```
Comment (evanlinjin) [inline PeriodicTableOfTests.md:1]: Not needed.

**Review by evanlinjin — CHANGES_REQUESTED** [review]
Everything that is not commented on by me, I completely agree with.

**Review by jdknives — CHANGES_REQUESTED** [review]
Needs to be merged into skywire-specs rather.

**Pull request closed without merging**